trin-core/src/portalnet/types.rs

```diff
@@ -705,4 +1026,412 @@ mod test {
         let hex_string = "500F";
         ProtocolId::from_str(hex_string).unwrap();
     }
+
+    // ==========================================
+    // Tests for the decoding of StateContentKeys
+    const RANDOM_HASH: &str = "b8be7903aee73b8f6a59cd44a1f52c62148e1f376c0dfa1f5f773a98666efc2b";
+    const RANDOM_HASH_2: &str = "d1c390624d3bd4e409a61a858e5dcc5517729a9170d014a6c96530d64dd8621d";
+    const RANDOM_HASH_3: &str = "3e190b68719aecbcb28ed2271014dd25f2aa633184988eb414189ce0899cade5";
+    const RANDOM_ADDRESS: &str = "829bd824b016326a401d083b33d092293333a830";
+
+    #[test]
+    fn test_account_trie_node_key_decode() {
+        let node_hash = vec_to_array(hex::decode(RANDOM_HASH).unwrap()).unwrap();
+        let state_root = vec_to_array(hex::decode(RANDOM_HASH_2).unwrap()).unwrap();
+
+        let container = AccountTrieNode {
+            path: VariableList::from(vec![1, 2, 0, 1]),
+            node_hash,
+            state_root,
+        };
+        let test_key =
+            ContentKey::StateContentKey(StateContentKey::AccountTrieNodeKey(container.clone()));
+
+        let ssz_bytes = test_key.to_bytes();
+        println!("{:?}", ssz_bytes);
+        let decoded = StateContentKey::from_bytes(&ssz_bytes).unwrap();
+        println!("{:?}", decoded);
+
+        if let StateContentKey::AccountTrieNodeKey(k) = decoded {
```

Comment:
I don't think this is ideal for our tests. It seems possible that these might not run at all if the `let` statement doesn't match. At the very least you should add an `else { panic!(..) }` statement to the end. Or you could do something like [here](https://stackoverflow.com/questions/34953711/unwrap-inner-type-when-enum-variant-is-known) where you implement functions that'll return the inner value

Reply:
A general testing question: we're expecting this test to succeed, since I made sure to have the proper input, but is including a panic just for safekeeping, in case the input was wrong somehow?

Reply:
Also, I just reviewed the code again and I think adding an `else { panic!(..) }` at the end of the `if` would be redundant because the `unwrap()` above will cause the test to panic in the case of decoding failure, which is what the else would be doing. The `if let` statement handles the case of successful decoding.

Reply:
Yeah including the panic is just for safekeeping. In this example, let's say the deserializing code changes somehow that instead of returning `AccountTrieNodeKey` for `decoded` - it returns `ContractStorageTrieNodeKey`. The `unwrap()` wouldn't catch this, b/c it was successfully deserialized. Then the if statement never gets run and the test passes successfully. Generally, I think it's smart to avoid `if` statements or other logic in your tests unless it's strictly necessary. In this scenario, adding a little `else {panic!()}` clause guarantees that either the tested assumptions inside the `if()` statement are run - or it panics.